lib/routes/novel/ptwxz.js

```diff
@@ -24,24 +25,57 @@ module.exports = async (ctx) => {
 
     const title = $('span>h1').text();
 
-    const list = $('.grid>tbody>tr>td>li>a');
+    const list = $('.grid>tbody>tr>td>li');
+
+    const chapter_item = list
+        .find('a')
+        .map((_, e) => ({
+            title: e.children[0].data,
+            link: e.attribs.href,
+            // pubDate // no date info found
+        }))
+        .get();
+
+    const items = await asyncPool(3, chapter_item, async (item) => {
+        const cache = await ctx.cache.get(item.link);
+        if (cache) {
+            return Promise.resolve(JSON.parse(cache));
+        }
+
+        const response = await got({
+            method: 'get',
+            url: `https://www.ptwxz.com/${item.link}`,
+            responseType: 'buffer',
+        });
+
+        const responseHtml = iconv.decode(response.data, 'GBK');
+        const $ = cheerio.load(responseHtml);
+
+        // remove all unwanted elements
+        // TODO: cleaner solution to rip text from body
+        $('div').remove();
+        $('h1').remove();
+        $('table').remove();
+        $('center').remove();
+        $('table').remove();
+        $('script').remove();
+
+        const description = $('body').html();
+
+        const single = {
+            title: item.title,
+            description,
+            link: item.link,
+        };
+        ctx.cache.set(item.link, JSON.stringify(single));
+        return Promise.resolve(single);
+    });
```

Comment:
```suggestion
    const items = await Promise.all(
        chapter_item.map((item) =>
            ctx.cache.tryGet(item.link, async () => {
                const response = await got({
                    method: 'get',
                    url: `https://www.ptwxz.com/${item.link}`,
                    responseType: 'buffer',
                });

                const responseHtml = iconv.decode(response.data, 'GBK');
                const $ = cheerio.load(responseHtml);

                // remove all unwanted elements
                // TODO: cleaner solution to rip text from body
                $('div').remove();
                $('h1').remove();
                $('table').remove();
                $('center').remove();
                $('table').remove();
                $('script').remove();

                const description = $('body').html();

                const single = {
                    title: item.title,
                    description,
                    link: item.link,
                };
                return single;
            })
        )
    );
```

Reply:
Removed `asyncPool` since the target site looks like not having any rate-limiting.
You can remove importing `tiny-async-pool`.

Reply:
got this part sorted out in new commits, still havent got the pubData working since I dont know how to retrieve `tiptitle` from this element


I guess a replace() like `replace('*更新时间:', '')`

Really not familar with js and regex so this is porbably a bit dumb, but i guess pubData is optional?
```js
    var d = // replace('*更新时间:', '')';
    var d1=d.split(" ");
    var date=d1[0].split("-");
    var time=d1[1].split(":");
    var mm=date[0];
    var dd=date[1];
    var hh=time[0];
    var min=time[1];
    pubData = new Date("2021", mm, dd, hh, mm);
```


```
<a href="/html/13/13534/9531977.html" target="_blank" tiptitle="305【同姓异迁】 更新时间:10-02 16:49">305【同姓异迁】</a>
```

```js
    const chapter_item = list
        .find('a')
        .map((_, e) => ({
            title: e.children[0].data,
            link: e.attribs.href,
            pubData: e.attribs.tiptitle // error undefined
        }))
        .get();
```

Reply:
const chapterList = $('#content li a')
        .map((_, item) => {
            item = $(item);
            const date = item.attr('tiptitle').match(/更新时间:(.*)/)[1]; //"10-04 00:17"
            return {
                title: item.text(),
                link: item.attr('href'),
                pubDate: parseDate(date, 'MM-DD HH:mm'),
            };
        })
        .get();

Reply:
method like `item.attr('tiptitle')` or `e.attribs.tiptitle` cant not retrieve `tiptitle="305【同姓异迁】 更新时间:10-02 16:49"`, onlt returns undefined.

Reply:
try with the attribute `title`